activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -484,13 +485,26 @@ def find_nth!(index)
       find_nth(index, offset_index) or raise RecordNotFound.new("Couldn't find #{@klass.name} with [#{arel.where_sql(@klass.arel_engine)}]")
     end
 
-    def find_nth_with_limit(offset, limit)
-      relation = if order_values.empty? && primary_key
-                   order(arel_table[primary_key].asc)
-                 else
-                   self
-                 end
+    # TODO: Decide how users' models should declare their natural ordering.
+    # Override this method? Declare a scope? Declare a set of fields with
+    # `#order` arg style?
+    def natural_ordering
+      if primary_key
+        order arel_table[primary_key].asc
+      else
+        raise "No natural ordering on #{name} since it has no primary key. Override #{name}.natural_ordering to provide your own, or use e.g. #take instead of #first."
+      end
+    end
+
+    def ordered
+      if order_values.empty?
```

Comment:
Okay, sorry, was just interested if there was a specific style guide related to this.